### PR TITLE
ENH add monitor keyword to the autoencoder

### LINKED ISSUE #20: add monitor keyword to the autoencoder

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds the monitor keyword to the autoencoder. 

Closes #20

### CODE REVIEW

**Comment by beckermr** [comment]
@samcarlos This one is good to go.

File: muffnn/autoencoder/autoencoder.py [hunk 1]
```diff
@@ -516,13 +516,22 @@ def _build_sigmoid_and_softmax_indices(self):
             raise ValueError("Sigmoid indices and softmax indices cannot"
                              " overlap!")
 
-    def partial_fit(self, X, y=None):
+    def partial_fit(self, X, y=None, monitor=None, **kwargs):
```
Comment (samcarlos) [inline muffnn/autoencoder/autoencoder.py]: Why do you pass kwargs here? Is there anything in partial_fit that takes that as an input?

Comment (beckermr) [reply muffnn/autoencoder/autoencoder.py]: Thus is part of the sklearn API. See here http://scikit-learn.org/stable/modules/generated/sklearn.base.TransformerMixin.html#sklearn.base.TransformerMixin.fit_transform

**Comment by beckermr** [comment]
@samcarlos This is fixed! Back to you!

**Pull request merged**